packages/qwik/src/optimizer/src/plugins/vite.ts

```diff
@@ -1119,30 +1119,48 @@ function absolutePathAwareJoin(path: Path, ...segments: string[]): string {
 export function convertManifestToBundleGraph(manifest: QwikManifest): QwikBundleGraph {
   const bundleGraph: QwikBundleGraph = [];
   const graph = manifest.bundles;
-  const map = new Map<string, { index: number; deps: string[] }>();
-  for (const bundleName in graph) {
+  if (!graph) {
+    return [];
+  }
+  const names = Object.keys(graph).sort();
+  const map = new Map<string, { index: number; deps: Set<string> }>();
+  const clearTransitiveDeps = (deps: Set<string>, seen: Set<string>, bundleName: string) => {
```

Comment:
changing this into a `function` instead of a `const` and moving it further down (after it gets called) will be better for readability I believe

Reply:
Actually I prefer it the other way round, I don't like how `function` looks and I like to know what a function does before it gets called...

Reply:
I understand. the reason I like the convention of "top down" is because it's easier this way to understand the flow of the whole thing before reading the specific implementation details.

This could start getting really hard to follow especially when dealing with longer functions you need to pass all the way just to figure out what's the main function is actually doing. 

We can move this discussion to discord and maybe create a coding style guide for the repo based on our decisions